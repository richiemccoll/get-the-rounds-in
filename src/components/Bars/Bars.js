import React from "react";
import { Dropdown } from "semantic-ui-react";
import PropTypes from "prop-types";

const Bars = ({ bars, selectBar }) => {
  return <Dropdown
          placeholder="Select bar.."
          fluid
          selection
          search
          options={bars}
          onChange={selectBar}
        />
};

export default Bars;

Bars.PropTypes = {
  bars: PropTypes.array.isRequired,
  selectBar: PropTypes.func.isRequired
};
