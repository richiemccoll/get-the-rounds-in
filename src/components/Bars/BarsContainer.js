import React, { Component } from "react";
import Bars from "./Bars";
import PropTypes from "prop-types";

class BarsContainer extends Component {
  constructor(props) {
    super(props);
    this.selectBar = this.selectBar.bind(this);
  }

  selectBar(event, data) {
    event.preventDefault();
    this.props.selectBar(data.options[data.value])
  }

  render() {
    const bars = this.props.bars.map(bar => {
      const { text, value } = bar;
      return { text, value }
    });

    return (
      <div>
        <Bars bars={bars} selectBar={this.selectBar} />
      </div>
    );
  }
}

export default BarsContainer;

BarsContainer.PropTypes = {
  selectBar: PropTypes.func,
  bars: PropTypes.array
};
