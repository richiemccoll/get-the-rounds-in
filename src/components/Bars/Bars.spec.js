import React from "react";
import Bars from "./Bars";
import BarsContainer from "./BarsContainer";
import renderer from "react-test-renderer";

describe("Bars", () => {
  const bars = [
    {
      id: 1,
      image_url: "./potstill.jpg",
      name: "The Pot Still",
      address: "154 Hope Street",
      description: "A real local favourite, receiving more shout-outs from our followers than anywhere else. It’s a small pub with a massive drink selection, particularly if you’re in the mood for a whisky. ",
      lat: 0,
      long: 0,
      text: "The Pot Still",
      value: "0"
    },
    {
      id: 2,
      image_url: "./stereo.png",
      name: "Stereo & The Old Hairdressers",
      address: "Renfield Lane",
      description: "These are actually two separate bars, but they have the same vibe, same owner and are right next to each other. Both have a chilled out, stripped back aesthetic and well stocked bars.",
      lat: 0,
      long: 0,
      text: "Stereo & The Old Hairdressers",
      value: "1"
    }
  ];

  const selectBar = () => console.log("selecting bar");

  it("should render the Bars component and pass the bars and selectBar function down as props", () => {
    const component = renderer.create(
      <BarsContainer bars={bars} selectBar={selectBar} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should display the bars correctly", () => {
    const component = renderer.create(
      <Bars bars={bars} selectBar={selectBar} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
