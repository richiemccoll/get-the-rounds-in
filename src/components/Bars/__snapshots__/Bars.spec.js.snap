// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bars should display the bars correctly 1`] = `
<div
  aria-busy={undefined}
  aria-disabled={undefined}
  aria-expanded={false}
  className="ui fluid search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
  tabIndex={undefined}
>
  <select
    aria-hidden="true"
    multiple={undefined}
    name={undefined}
    type="hidden"
    value=""
  >
    <option
      value=""
    />
    <option
      value="0"
    >
      The Pot Still
    </option>
    <option
      value="1"
    >
      Stereo & The Old Hairdressers
    </option>
  </select>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    name="-search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    className="default text"
  >
    Select bar..
  </div>
  <i
    aria-hidden="true"
    className="dropdown icon"
  />
  <div
    aria-multiselectable={undefined}
    className="menu transition"
    role="listbox"
  >
    <div
      address="154 Hope Street"
      aria-checked={false}
      aria-disabled={undefined}
      aria-selected={true}
      className="selected item"
      id={1}
      image_url="./potstill.jpg"
      lat={0}
      long={0}
      name="The Pot Still"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="description"
      >
        A real local favourite, receiving more shout-outs from our followers than anywhere else. It’s a small pub with a massive drink selection, particularly if you’re in the mood for a whisky. 
      </span>
      <span
        className="text"
      >
        The Pot Still
      </span>
    </div>
    <div
      address="Renfield Lane"
      aria-checked={false}
      aria-disabled={undefined}
      aria-selected={false}
      className="item"
      id={2}
      image_url="./stereo.png"
      lat={0}
      long={0}
      name="Stereo & The Old Hairdressers"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="description"
      >
        These are actually two separate bars, but they have the same vibe, same owner and are right next to each other. Both have a chilled out, stripped back aesthetic and well stocked bars.
      </span>
      <span
        className="text"
      >
        Stereo & The Old Hairdressers
      </span>
    </div>
  </div>
</div>
`;

exports[`Bars should render the Bars component and pass the bars and selectBar function down as props 1`] = `
<div
  aria-busy={undefined}
  aria-disabled={undefined}
  aria-expanded={false}
  className="ui fluid search selection dropdown"
  onBlur={[Function]}
  onChange={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseDown={[Function]}
  role="combobox"
  tabIndex={undefined}
>
  <select
    aria-hidden="true"
    multiple={undefined}
    name={undefined}
    type="hidden"
    value=""
  >
    <option
      value=""
    />
    <option
      value="0"
    >
      The Pot Still
    </option>
    <option
      value="1"
    >
      Stereo & The Old Hairdressers
    </option>
  </select>
  <input
    aria-autocomplete="list"
    autoComplete="off"
    className="search"
    name="-search"
    onChange={[Function]}
    style={
      Object {
        "width": undefined,
      }
    }
    tabIndex={0}
    type="text"
    value=""
  />
  <div
    className="default text"
  >
    Select bar..
  </div>
  <i
    aria-hidden="true"
    className="dropdown icon"
  />
  <div
    aria-multiselectable={undefined}
    className="menu transition"
    role="listbox"
  >
    <div
      aria-checked={false}
      aria-disabled={undefined}
      aria-selected={true}
      className="selected item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        The Pot Still
      </span>
    </div>
    <div
      aria-checked={false}
      aria-disabled={undefined}
      aria-selected={false}
      className="item"
      onClick={[Function]}
      role="option"
      style={
        Object {
          "pointerEvents": "all",
        }
      }
    >
      <span
        className="text"
      >
        Stereo & The Old Hairdressers
      </span>
    </div>
  </div>
</div>
`;
